hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/ExperimentalTpcAuthenticationMessageTask.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.task;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.ExperimentalTpcAuthenticationCodec;
+import com.hazelcast.instance.impl.Node;
+import com.hazelcast.internal.nio.Connection;
+
+import java.security.Permission;
+
+public class ExperimentalTpcAuthenticationMessageTask
+        extends AbstractMessageTask<ExperimentalTpcAuthenticationCodec.RequestParameters> {
+
+    public ExperimentalTpcAuthenticationMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
+        super(clientMessage, node, connection);
+    }
+
+    @Override
+    protected ExperimentalTpcAuthenticationCodec.RequestParameters decodeClientMessage(ClientMessage clientMessage) {
+        return null;
```

Comment:
Is returning `null` intentional here?

Reply:
Yes. We don't need to decode the message here again, as we don't need to do anything in this message task other than to send a null response.